qa-deploy

```diff
@@ -115,9 +115,14 @@ function run() {
     values_file="$2"
     tag_to_deploy=${3:-latest}
 
-    # Mandatory arguments
-    if [ -z "${environment:-}" ]; then invalid "No environment specified (e.g. 'production')"; fi
-    if [ -z "${values_file:-}" ]; then invalid "No values file specified (e.g. 'sites/canonical-com.yaml')"; fi
+    # Validate arguments
+    if [[ ! "$environment" =~ ^(production|staging)$ ]]; then
+        invalid "You need to specify a valid environment: 'production' or 'staging'"
+    fi
+
+    if [[ ! -f $values_file ]] ; then
+        invalid "File "${values_file}" doesn't exits, aborting."
```

Comment:
If you simply don't provide the second argument, won't this alert an error like:

> File "" doesn't exits, aborting.

This seems worse than the existing "No values file specified" message. Maybe we should do both tests?

Also, I think you need to change the `"` to `'`

Reply:
Hey @nottrobin , at the beginning of the script there was a `set -euo` so the previous conditions to check if empty didn't work anyway.
```
$ ./qa-deploy
./qa-deploy: line 114: $1: unbound variable
```

Please review again, I decided to remove the -u flag in the set command and add back these two conditions to check if the arguments are empty because then the user will have a nice output either they leave it empty or they misuse the arguments.